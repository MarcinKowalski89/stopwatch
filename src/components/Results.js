import React from 'react';
import PropTypes from 'prop-types';

const Results = props => (
  <div>
    <h2>List of time results</h2>
    <ul className="list-group">
      {props.items.map((item, i) => {
        const key = i + 1;
        return (
          <li key={key} className="list-group-item">{i + 1}&nbsp;-&nbsp;
            <span>{item.getHours() < 10 ? `0${item.getHours()}` : item.getHours()}:</span>
            <span>{item.getMinutes() < 10 ? `0${item.getMinutes()}` : item.getMinutes()}:</span>
            <span>{item.getSeconds() < 10 ? `0${item.getSeconds()}` : item.getSeconds()}.</span>
            <span>{item.getMilliseconds()}</span>
          </li>
        );
      })}
    </ul>
  </div>
);

Results.propTypes = { items: PropTypes.array.isRequired };

export default Results;
