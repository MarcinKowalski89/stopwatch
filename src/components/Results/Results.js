import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Time from '../Time/Time';

export const Results = props => (
  <div>
    <h2>List of time results</h2>
    <ul className="list-group">
      {props.results.map((result, i) => {
        const key = i + 1;
        return (
          <li key={key} className="list-group-item">{i + 1}&nbsp;-&nbsp;
            <Time time={result} />
          </li>
        );
      })}
    </ul>
  </div>
);

Results.propTypes = { results: PropTypes.arrayOf(Date).isRequired };

export default connect(state => ({ results: state.stopwatch.results }))(Results);
