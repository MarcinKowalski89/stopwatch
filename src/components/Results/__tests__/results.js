import React from 'react';
import { shallow } from 'enzyme';

import { Results } from '../Results';

test('Results shoud match default snapshot', () => {
  const component = shallow(<Results
    results={[]}
  />);

  expect(component).toMatchSnapshot();
});

test('Results shoud match snapshot for 3 results', () => {
  const component = shallow(<Results
    results={[new Date('2017-06-13T04:41:20'), new Date('2017-06-13T04:42:20'), new Date('2017-06-13T04:43:20')]}
  />);

  expect(component).toMatchSnapshot();
});
