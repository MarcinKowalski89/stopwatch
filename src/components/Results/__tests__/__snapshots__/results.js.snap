// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results shoud match default snapshot 1`] = `
<div>
  <h2>
    List of time results
  </h2>
  <ul
    className="list-group"
  />
</div>
`;

exports[`Results shoud match snapshot for 3 results 1`] = `
<div>
  <h2>
    List of time results
  </h2>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
      key="1"
    >
      1
       - 
      <Time
        time={2017-06-13T04:41:20.000Z}
      />
    </li>
    <li
      className="list-group-item"
      key="2"
    >
      2
       - 
      <Time
        time={2017-06-13T04:42:20.000Z}
      />
    </li>
    <li
      className="list-group-item"
      key="3"
    >
      3
       - 
      <Time
        time={2017-06-13T04:43:20.000Z}
      />
    </li>
  </ul>
</div>
`;
